### PR TITLE
style: Added relevant icons for preferences UI #4902

### LINKED ISSUE #4902: Add suitable icons in `Preferences`
### What
Add suitable icons in `Preferences`.

### Why/User story
Add suitable icons in `Preferences` for better UI.

### Before

off

### PR DESCRIPTION
**Pull request by anshgovil**
####  Description
Added appropriate icons (SVG) to the preferences fragment UI. SVGs (most) used (and edited) from svgrepo. Appropriate licensing was checked to allow change and use of the SVG.

#### Related issues
Fixes #4902 

#### Screenshots
p1
p2
p3
p4

### CODE REVIEW

**Comment by anshgovil** [comment]
@VaiTon Can you review this PR? 
(11/12 checks passed with the fail test being outdated dependencies in gradle prevalent from develop branch)

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/features/preferences/PreferencesFragment.kt [hunk 1]
```diff
@@ -415,7 +418,11 @@ class PreferencesFragment : PreferenceFragmentCompat(), INavigationItem {
 
         key = config.type
         title = getString(R.string.display_analysis_tag_status, config.typeName.lowercase(Locale.getDefault()))
-
+        when (config.typeName.lowercase(Locale.getDefault())) {
+            "palm oil" -> icon = ContextCompat.getDrawable(requireContext(), R.drawable.ic_preference_drop)
+            "vegan" -> icon = ContextCompat.getDrawable(requireContext(), R.drawable.ic_preference_vegan)
+            "vegetarian" -> icon = ContextCompat.getDrawable(requireContext(), R.drawable.ic_preference_vegetarian)
+        }
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/features/preferences/PreferencesFragment.kt]: Consider doing something like
```
val drawable = when(...){
    ... -> R.drawable. ...
}
icon = ContextCompat.getDrawable(requireContext(), drawable)
```

**Review by VaiTon — APPROVED** [review]
Just a quick remark. Also, I would remove the icons from the section headers, but that's only my style preference. Code-wise OK!

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/features/preferences/PreferencesFragment.kt (hunk 1 above)
Comment (anshgovil) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/features/preferences/PreferencesFragment.kt]: Updated code to match code style mentioned + removed header icons in the latest commit.

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/features/preferences/PreferencesFragment.kt [hunk 2]
```diff
@@ -418,11 +418,13 @@ class PreferencesFragment : PreferenceFragmentCompat(), INavigationItem {
...
-            "palm oil" -> icon = ContextCompat.getDrawable(requireContext(), R.drawable.ic_preference_drop)
-            "vegan" -> icon = ContextCompat.getDrawable(requireContext(), R.drawable.ic_preference_vegan)
-            "vegetarian" -> icon = ContextCompat.getDrawable(requireContext(), R.drawable.ic_preference_vegetarian)
+        val drawable = when (config.typeName.lowercase(Locale.getDefault())) {
+            "palm oil" -> R.drawable.ic_preference_drop
+            "vegan" -> R.drawable.ic_preference_vegan
+            "vegetarian" -> R.drawable.ic_preference_vegetarian
+            else -> null
         }
+        icon = ContextCompat.getDrawable(requireContext(), drawable!!)
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/features/preferences/PreferencesFragment.kt]: Replace this with a 
```suggestion
drawable?.let {
        icon = ContextCompat.getDrawable(requireContext(), it)
}
```

we don't want to crash the app if we cannot find an icon

Comment (anshgovil) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/features/preferences/PreferencesFragment.kt]: My oversight; I assumed the icon will always be available in the app package or else nulled (which then checked). --Fixed now

**Review by VaiTon — APPROVED** [review]

**Comment by VaiTon** [comment]
Thank you! 🎉

**Comment by teolemon** [comment]
I'm a bit annoyed because they clearly lack consistency (eg the lines have a different thickness) with material icons which we use throughout the app

**Pull request merged**